test/integration/consumer_group_test.go

```diff
@@ -30,6 +34,317 @@ import (
+	require.NoError(t, err)
+	cleaner.Add(gwc)
+
+	t.Log("deploying a new gateway")
+	gatewayName := uuid.NewString()
+	gateway, err := helpers.DeployGateway(ctx, gatewayClient, ns.Name, gatewayClassName, func(gw *gatewayapi.Gateway) {
+		gw.Name = gatewayName
+	})
+	require.NoError(t, err)
+	cleaner.Add(gateway)
+
+	deployment, service, ingress, keyauthPlugin := deployMinimalSvcWithKeyAuth(ctx, t, ns.Name, path)
+	cleaner.Add(deployment)
+	cleaner.Add(service)
+	cleaner.Add(keyauthPlugin)
+
+	// This borrows a lot from TestConsumerGroup, since it wants to test the same things for the GWAPI case, and reusing
+	// test helpers simplifies that. deployMinimalSvcWithKeyAuth always creates an Ingress though, which we don't want.
+	// Deleting it immediately is a bit silly, but simpler than refactoring the whole helper to be more modular. We should
+	// consider such a refactor if we start using it more widely.
+	require.NoError(t, clusters.DeleteIngress(ctx, env.Cluster(), ns.Name, ingress))
+
+	t.Log("creating plugins to attach to the route and groups")
+	addedHeader := header{
+		K: "X-Test-Header",
+		V: "added",
+	}
+	// Use the same header key as plugin name.
+	pluginRespTrans := configurePlugin(
+		ctx, t, ns.Name, "response-transformer-advanced-1", "response-transformer-advanced", fmt.Sprintf(
+			`{
+				"add": {
+					"headers": [
+						"%s: %s"
+					]
+				}
+			 }`,
+			addedHeader.K, addedHeader.V,
+		),
+	)
+	cleaner.Add(pluginRespTrans)
+
+	const rateLimitValue = 100
+	pluginRateLimit := configurePlugin(
+		ctx, t, ns.Name, "rate-limiting-advanced-1", "rate-limiting-advanced", fmt.Sprintf(
+			`{
+				"limit": [%d],
+				"window_size": [100],
+				"namespace": "test",
+				"sync_rate": -1,
+				"window_type": "fixed"
+			 }`,
+			rateLimitValue,
+		),
+	)
+	cleaner.Add(pluginRateLimit)
+
+	t.Logf("creating an httproute to access deployment %s via kong", deployment.Name)
+	httpPort := gatewayapi.PortNumber(service.Spec.Ports[0].Port)
+	pathMatchPrefix := gatewayapi.PathMatchPathPrefix
+	httpRoute := &gatewayapi.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: uuid.NewString(),
+			Annotations: map[string]string{
+				annotations.AnnotationPrefix + annotations.StripPathKey: "true",
+				annotations.AnnotationPrefix + annotations.PluginsKey:   strings.Join([]string{keyauthPlugin.Name}, ","),
+			},
+		},
+		Spec: gatewayapi.HTTPRouteSpec{
+			CommonRouteSpec: gatewayapi.CommonRouteSpec{
+				ParentRefs: []gatewayapi.ParentReference{{
+					Name: gatewayapi.ObjectName(gateway.Name),
+				}},
+			},
+			Rules: []gatewayapi.HTTPRouteRule{{
+				Matches: []gatewayapi.HTTPRouteMatch{
+					{
+						Path: &gatewayapi.HTTPPathMatch{
+							Type:  &pathMatchPrefix,
+							Value: kong.String(path),
+						},
+					},
+				},
+				BackendRefs: []gatewayapi.HTTPBackendRef{{
+					BackendRef: gatewayapi.BackendRef{
+						BackendObjectReference: gatewayapi.BackendObjectReference{
```

Comment:
We can use builder for these entities and make call sites shorter: https://github.com/Kong/kubernetes-ingress-controller/blob/513db87cbf94ce66207f74365b502e8cde841357/internal/util/builder/backendref.go#L17